uudeview: don't hardeningDisable format

woffs opened the pull request:
###### Motivation for this change
- adopt a patch which several other distributions use
- name -> pname + version

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented:
Result of `nixpkgs-review pr 99403` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>uudeview</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/tools/misc/uudeview/default.nix:
```diff
@@ -1,18 +1,35 @@
-  patches = [ ./matherr.patch ];
+  patches = [
+    # https://wiki.tcl.tk/3577
+    ./matherr.patch
```
Why is this patch required? Renaming can be done with a command.

woffs replied:
that patch was introduced in 14fc982e0c03e3ef220a95a0f724b02d4c916a8f by @Infinisil and I do not touch it. I only modify the patch header to be compatible to patchFlags = [ "-p0" ] which I must set for the newly added patch

SuperSandro2000 commented:
Result of `nixpkgs-review pr 99403` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>uudeview</li>
  </ul>
</details>

Infinisil commented on pkgs/tools/misc/uudeview/default.nix:
```diff
@@ -1,18 +1,35 @@
+    })
+  ];
+
+  patchFlags = [ "-p0" ]; # needed for fetched patch
```
You don't need this flag or a change in the existing patch if you use `fetchpatch { ...; extraPrefix = ""; }` in the `fetchpatch` call (make sure to update the `sha256` when you do).

woffs replied:
> You don't need this flag or a change in the existing patch if you use `fetchpatch { ...; extraPrefix = ""; }` in the `fetchpatch` call (make sure to update the `sha256` when you do).

Great. Done.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 99403` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>uudeview</li>
  </ul>
</details>

`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

woffs committed:
uudeview: don't hardeningDisable format

- adopt a patch which several other distributions use
- name -> pname + version
- license: gpl2Plus
